integration-cli/docker_cli_ps_test.go

```diff
@@ -666,3 +666,58 @@ func (s *DockerSuite) TestPsGroupPortRange(c *check.C) {
 	}
 
 }
+
+// Test for GitHub issue #12595
+func (s *DockerSuite) TestPsImageIDAfterUpdate(c *check.C) {
+
+	originalImageName := "busybox:TestPsImageIDAfterUpdate-original"
+	updatedImageName := "busybox:TestPsImageIDAfterUpdate-updated"
```

Comment:
@thieman reading through this test I'm having a hard time telling if it's actually testing the problem I see. There could be more than one here. These are the steps that currently result in an error for us:

1. Image exists as `busybox`
2. I start up a container
3. `docker ps` shows my image as `busybox:latest`
3. `busybox` is outdated and I do a `docker pull` which updates the image
4. `docker ps` shows my image ID as *2485b0f8995167b796a7ab4696e842c08bd27d58febb3ba0e935a4e2c1b2807d:latest*